File: abundant_etc.js
#!/usr/bin/env node
// abundant, perfect, deficient numbers in js

function divisors(n) {
  var div = []
  for (var i = 0; i <= Math.floor(Math.sqrt(n)); i++) {
    if (n % i === 0) {
      div.push(i);
      if (n/i !== i) {
        div.push(n/i);
      }
    }
  }
  div.sort(function(x,y) { return x - y });
  return div;
}

function sumList(list,i,c) {
  if (i === undefined) {
    var i = 0;
  }
  if (c === undefined) {
    var c = 0;
  }
  if (i >= list.length) {
    return c;
  } else {
    return(sumList(list,i+1,c+list[i]));
  }
}

function sumDivisors(n) {
  var div = divisors(n);
  div.pop();
  return sumList(div);
}

function abundant(n) {
  return sumDivisors(n) > n;
}

function deficient(n) {
  return sumDivisors(n) < n;
}

function perfect(n) {
  return sumDivisors(n) === n;
}

for (var i = 1; i < 1000; i++) { 
  var kind = abundant(i) ? "abundant" : deficient(i) ? "deficient" : "perfect";
  console.log(i + ": " + kind);
}

File: abundant_etc.py
#!/usr/bin/env python

def divisors(n):
  div = list()
  for d in range(1,n-1):
    if n % d == 0:
      div.append(d)
  return div

def sum_list(list,index,accumulator):
  if index >= len(list):
    return accumulator
  else:
    return sum_list(list,index+1,accumulator+list[index])

def sum_list_wrapper(list):
  return sum_list(list,0,0)

def classify(n):
  result = sum_list_wrapper(divisors(n))
  if result < n:
    return -1
  elif result == n:
    return 0
  elif result > n:
    return 1

for i in range(1,100):
  kind = classify(i)
  if kind < 0:
    print str(i) + ": deficient" 
  elif kind == 0:
    print str(i) + ": perfect"
  elif kind > 0:
    print str(i) + ": abundant"
